### PR TITLE
ResizableBox: Remove knobs in stories

### PR DESCRIPTION
**Pull request by mirka**
Part of #35665

## What?

Remove the Knobs add-on from `ResizableBox` stories.

## Why?

This is now a blocker for supporting npm 8 (#46443).

## How?

Just a quick, minimal refactor.

## Testing Instructions

1. `npm run storybook:dev`
2. Go to the `ResizableBox` component.
3. The stories and controls should be covering the cases covered by the previous knobs. See inline code comments for intentional alterations.

### CODE REVIEW

File: packages/components/src/resizable-box/index.tsx [hunk 1]
```diff
@@ -124,4 +124,6 @@ function ResizableBox(
 	);
 }
 
-export default forwardRef( ResizableBox );
+const ResizableBox = forwardRef( UnforwardedResizableBox );
+
+export default ResizableBox;
```
Comment (mirka) [inline packages/components/src/resizable-box/index.tsx]: Tweaked the component export so we can use the TypeScript data in the Storybook props table.

File: packages/components/src/resizable-box/stories/index.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+			{ ...props }
+			size={ {
+				height,
+				width,
+			} }
+			onResizeStop={ ( event, direction, elt, delta ) => {
+				onResizeStop?.( event, direction, elt, delta );
+				setAttributes( {
+					height: height + delta.height,
+					width: width + delta.width,
```
Comment (mirka) [inline packages/components/src/resizable-box/stories/index.tsx:51]: The `parseInt()` that used to be here turned out to be unnecessary. Thanks TypeScript 🙏

File: packages/components/src/resizable-box/stories/index.js [hunk 3]
```diff
@@ -1,97 +0,0 @@
...
-				} }
-			>
-				{ children }
-			</ResizableBox>
-		</div>
-	);
-};
-
-export const _default = () => {
-	const content = text( 'Example: Content', 'Resize' );
```
Comment (mirka) [inline packages/components/src/resizable-box/stories/index.js:54]: `content` is not an actual component prop.

File: packages/components/src/resizable-box/stories/index.tsx [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+const meta: ComponentMeta< typeof ResizableBox > = {
+	title: 'Components/ResizableBox',
+	component: ResizableBox,
+	argTypes: {
+		__experimentalShowTooltip: { control: 'boolean' },
+		minHeight: { control: 'number' },
+		minWidth: { control: 'number' },
+		onResizeStop: { action: 'onResizeStop' },
+		showHandle: { control: 'boolean' },
```
Comment (ciampo) [inline packages/components/src/resizable-box/stories/index.tsx]: Do you know why are these props not shown by default in the Storybook docs?

**Review by ciampo — COMMENTED** [review]
Screenshot 2023-01-05 at 13 23 24

With respect to the version on trunk, the `enable` prop is not always showing (it's only showing in the specific storybook example) — should we look into having it always visible in the component controls?

**Comment by mirka** [comment]
> With respect to the version on trunk, the `enable` prop is not always showing (it's only showing in the specific storybook example) — should we look into having it always visible in the component controls?

✅ Done in 7cd7eab

File: packages/components/src/resizable-box/stories/index.tsx (hunk 4 above)
Comment (mirka) [reply packages/components/src/resizable-box/stories/index.tsx]: Good question. I tweaked the export so at least our first-party types are visible ✅ 

However, all the props coming from the upstream [`re-resizable`](https://github.com/WordPress/gutenberg/blob/d3eaa6cacb4038fd2cab7f49ef4444e64ba6e6eb/packages/components/src/resizable-box/index.tsx#L90) are not shown at all 🤔 So I think what will happen is that in the full TypeScript migration, we'll decide which ones of the upstream props are noteworthy, and add our own typings/descriptions for them so they'll be shown in the Storybook props table. For all the other upstream props, the main component description will nudge people toward the `re-resizable` docs as written in the [readme](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/resizable-box/README.md).

File: packages/components/src/resizable-box/stories/index.tsx [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+
+const meta: ComponentMeta< typeof ResizableBox > = {
+	title: 'Components/ResizableBox',
+	component: ResizableBox,
+	argTypes: {
```
Comment (ciampo) [inline packages/components/src/resizable-box/stories/index.tsx:19]: We should probably hide `children`'s control, as it currently shows a react object that can't really be edited

**Review by ciampo — APPROVED** [review]
One last smaller comment and then you can go ahead and merge this one too 🚀

File: packages/components/src/resizable-box/stories/index.tsx (hunk 4 above)
Comment (ciampo) [reply packages/components/src/resizable-box/stories/index.tsx]: Sounds like a plan

File: packages/components/src/resizable-box/stories/index.tsx (hunk 5 above)
Comment (mirka) [reply packages/components/src/resizable-box/stories/index.tsx:19]: True 👍

**Pull request merged**